src/common.ts

```diff
@@ -0,0 +1,58 @@
+import { addMinutes, set } from 'date-fns';
+import type { z } from 'zod';
+
+export function startOfTheNextMinute(fromDate?: Date): Date {
+  const now = fromDate ?? new Date();
+
+  const zeroSeconds = set(now, {
+    seconds: 0,
+    milliseconds: 0,
+  });
+
+  return addMinutes(zeroSeconds, 1);
+}
+
+/**
+ *
+ * @param data search parameters
+ * @returns the parameters formatted as URLSearchParams
+ *
+ * @example  makeSearchParams({ foo: 'bar', baz: 42 }) => 'foo=bar&baz=42'
+ *
+ * Create search parameters from a flat object.
+ */
+/** Creates search params from a one-dimensional object. Supports array values. */
+export function makeSearchParams(
+  data: Record<string, unknown>,
+): URLSearchParams {
+  const params = new URLSearchParams();
+  Object.entries(data).forEach(([key, value]) => {
+    if (value) {
+      if (Object.prototype.toString.call(value) === '[object Array]') {
+        Object.values(value).forEach(v => {
+          params.append(key, makeStringFromSeachParam(v));
+        });
+      } else {
+        params.append(key, makeStringFromSeachParam(value));
+      }
+    }
+  });
+
+  return params;
+}
+
+function makeStringFromSeachParam(value: unknown): string {
+  if (value instanceof Date) {
+    return value.toISOString();
+  }
```

Comment:
This is a change in behaviour, not merely a refactoring. It seems the existing codebase uses `format(date, 'yyyy-MM-dd')`.

[Abax' documentation](https://developers.abax.cloud/documentation/api#/Trips/getTripsList) describes slightly differently shaped dates, which include time. Perhaps we should follow that as closely as possible. However, that is a change for another PR, not this refactor.

Reply:
fixed by using format(date, 'yyyy-mm-ff')